src/Middleware/CheckIfTokenIsFresh.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelCom\AuthModule\Middleware;
+
+use Carbon\Carbon;
+use Closure;
+use Illuminate\Http\Request;
+use MyParcelCom\AuthModule\JwtRequestAuthenticator;
+use MyParcelCom\JsonApi\Exceptions\InvalidAccessTokenException;
+
+readonly class CheckIfTokenIsFresh
```

Comment:
@yoan-myparcel pointed out to me once that `readonly` classes cannot be mocked, because ["`readonly` classes cannot be extended by classes that are not `readonly`"](https://docs.phpunit.de/en/10.5/test-doubles.html). 

Not sure how likely it is that we would want to mock a middleware, but we can easily keep that opportunity by making the constructor argument `readonly`, instead if the class